collectors/o365/o365_collector.js

```diff
@@ -164,6 +164,11 @@ class O365Collector extends PawsCollector {
 
             return callback(null, logs, newState, newState.poll_interval_sec);
         }).catch(err => {
+            if(err.message){
+                let message = err.message;
+                let error = JSON.parse(message.slice(message.indexOf('{'), message.lastIndexOf('}') + 1));
```

Comment:
I'm sure youve got a good reason for doing this, but that looks fragile to me. And considering its an MS api, it might change with little warning.

I would wrap this line in a try/catch block and return some sort of generic error if that parsing fails

Reply:
wrap the paring error and pushed the changes